kafka-ui-api/src/main/java/com/provectus/kafka/ui/serdes/builtin/ProtobufFileSerde.java

```diff
@@ -70,9 +78,33 @@ public void configure(PropertyResolver serdeProperties,
   }
 
   private void configure(PropertyResolver properties) {
-    Map<Path, ProtobufSchema> protobufSchemas = joinPathProperties(properties).stream()
-        .map(path -> Map.entry(path, new ProtobufSchema(readFileAsString(path))))
-        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    List<Location> locations = joinPathProperties(properties).stream()
+        .map(ProtobufFileSerde::toLocation)
+        .filter(Objects::nonNull)
+        .distinct()
+        .toList();
+
+    SchemaLoader schemaLoader = new SchemaLoader(FileSystems.getDefault());
+    schemaLoader.setLoadExhaustively(true);
+    schemaLoader.setPermitPackageCycles(true);
+    schemaLoader.initRoots(locations, List.of());
+
+    final Schema schema;
+    try {
+      schema = schemaLoader.loadSchema();
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
```

Comment:
please extract to static package-access `ProtobufFileSerde::loadProtobufSchema` method and use it in tests also. Maybe put `toLocation` logic there too

Reply:
Done.

https://github.com/provectus/kafka-ui/pull/3039/commits/b7a91fbd9f37930f56aa9d1fde1bf33af5675ec0